### DISCUSSION TITLE
Socket.IO Connection

### DISCUSSION THREAD

**Discussion by ghost** [homebridge-simplisafe3/homebridge-simplisafe3#310] [Q&A] [answered]
First off thank you for all the hard work making this API wrapper!

I'm currently working on building this wrapper for Arduino to work with [HomeSpan](https://github.com/HomeSpan/HomeSpan). I'm having a hard time getting the Socket.IO connection working though, I can't find SimpliSafe's implementation anywhere in any of their apps. Your implementation obviously works but there's no official Socket.IO library for Arduino so I'm having to piece together my own. 

Can you point me to where you found this endpoint and how you went about reverse engineering it? I understand it's been in this repo since V1 so maybe SimpliSafe used to use it and doesn't anymore? If that's the case are you concerned about them supporting it long term? 

Thanks again for this awesome project!

**Comment by shamoon** [comment, marked as answer]
Hi @codysorgenfrey1 , I did not originally create the API wrapper (thus did not reverse-engineer it myself), not sure if it used to be used by the website or the app. The other place to look (i.e. maybe ask) is https://github.com/bachya/simplisafe-python btw

Im not aware of them deprecating this API but of course its a concern in the future—always!

**Reply by ghost** [reply]
Thanks for the tip! 

I'm no expert at Python but it looks like they're actually using a plain WebSocket at `wss://socketlink.prd.aser.simplisafe.com`. I did actually see this endpoint on [SimpliSafe's current web app](https://webapp.simplisafe.com). This might be the best option if in the future they yank the plug on the socket.io server.

**Reply by shamoon** [reply]
Ps. if youre curious in the latest version I updated to the new WebSocket =)

**Reply by ghost** [reply]
I also successfully got this WebSocket working for my implementation on Arduino! It had an added bonus of not having to work on implementing SocketIO for Arduino with namespace support. 

Thanks for the tip!
